enkit/outputs/commands.go

```diff
@@ -55,26 +60,83 @@ func NewRoot(base *client.BaseFlags) (*Root, error) {
 type Mount struct {
 	*cobra.Command
 	root *Root
+
+	BuildBuddyApiKey  string
+	BuildBuddyUrl     string
+	ClusterName       string
+	DryRun            bool
+	InvocationID      string
+	SymlinkFormat     string
+	UseTestResults    bool
+	UseBuildArtifacts bool
 }
 
 func NewMount(root *Root) *Mount {
 	command := &Mount{
 		Command: &cobra.Command{
-			Use:   "mount [invocation ID]",
+			Use:   "mount",
 			Short: "Mount the build outputs of a particular invocation",
 			Example: `  $ enkit outputs mount 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
 	Mounts outputs from build 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07 to the
 	default location.`,
-			Args: cobra.ExactArgs(1),
 		},
 		root: root,
 	}
+	command.Flags().StringVar(&command.BuildBuddyApiKey, "api-key", "", "build buddy api key used to bypass oauth2")
+	command.Flags().StringVar(&command.BuildBuddyUrl, "url", "", "build buddy url instance")
+	command.Flags().StringVar(&command.ClusterName, "cluster", "", "name of the cluster")
+	command.Flags().StringVarP(&command.InvocationID, "invocation", "i", "", "invocation id to mount")
+	command.Flags().BoolVarP(&command.UseTestResults, "test-artifacts", "t", true, "mount test artifacts")
+	command.Flags().BoolVarP(&command.UseBuildArtifacts, "build-artifacts", "b", false, "mount build artifacts")
+	command.Flags().BoolVar(&command.DryRun, "dry-run", false, "if set, will print out the symlinks generated from the invocation, and not attempt to create them")
+
 	command.Command.RunE = command.Run
 	return command
 }
 
 func (c *Mount) Run(cmd *cobra.Command, args []string) error {
-	return fmt.Errorf("`enkit outputs mount` is unimplemented")
+	buddyUrl, err := url.Parse(c.BuildBuddyUrl)
+	if err != nil {
+		return err
+	}
+	bc, err := bes.NewBuildBuddyClient(buddyUrl, c.root.BaseFlags, c.BuildBuddyApiKey)
+	if err != nil {
+		return err
+	}
+	var mods []kbuildbarn.FilterOption
+	if c.UseTestResults {
+		fmt.Println("mounting with test artifacts")
+		mods = append(mods, kbuildbarn.WithTestResults())
+	}
+	if c.UseBuildArtifacts {
+		fmt.Println("mounting with build artifacts")
+		mods = append(mods, kbuildbarn.WithNamedSetOfFiles())
+	}
+	r, err := kbuildbarn.GenerateSymlinks(context.Background(), bc, c.root.OutputsRoot, c.InvocationID, c.ClusterName, mods...)
+	if err != nil {
+		return err
+	}
+	if err := os.Mkdir(filepath.Join(c.root.OutputsRoot, "scratch", c.InvocationID), 0777); err != nil && !os.IsExist(err) {
```

Comment:
At some point in here we need to start bb_clientd if it wasn't already running, and/or detect a running instance of bb_clientd - probably should fetch the scratch directory from a bb_clientd handle, instead of inferring it here.

TODO is fine, since this library doesn't exist yet

Reply:
Case in point - scratch dir is actually in `<OutputsRoot>/.bb_clientd/scratch`

Reply:
Added in the TODO